frontend/src/home/EditPlanPopper.tsx

```diff
@@ -139,24 +139,20 @@ export class EditPlanPopperComponent extends React.Component<
   onChooseMajor(event: React.SyntheticEvent<{}>, value: any) {
     const maj = this.props.majors.find((m: any) => m.name === value);
     this.props.setMajor(maj);
-    this.props.setCoopCycle(undefined);
+    this.props.setCoopCycle("");
   }
 
   /**
    * Updates this user's plan based on the plan selected in the dropdown.
    */
   onChoosePlan(event: React.SyntheticEvent<{}>, value: any) {
-    if (value === "None") {
-      this.props.setCoopCycle(undefined);
-      return;
-    }
-
-    const plan = this.props.plans[this.props.major!.name].find(
+    const plan = this.props.plans[this.props.major].find(
       (p: Schedule) => planToString(p) === value
     );
 
     if (plan) {
-      this.props.setCoopCycle(plan);
+      const chosenCoopCycle = value === "None" ? "" : value;
+      this.props.setCoopCycle(chosenCoopCycle, plan);
```

Comment:
Do we want users to be able to unset coopCycle? In this case would I need to send `undefined` instead of an empty string if None was selected?

Reply:
Yeah we normally keep coopCycle as an optional field (coopCycle?: string is the same as coopCycle: string | undefined)